### DISCUSSION TITLE
Alternative Title Database

### DISCUSSION THREAD

**Discussion by SporyTike** [RetroAchievements/RAWeb#1861] [Ideas]
A long time issue on the site is the game name handling, usually we go with the idea of writing title `<GameTitle1> | <GameTitle2>` like you can see here:
![image](https://github.com/RetroAchievements/RAWeb/assets/45672834/172fc4d4-61af-4cb4-a587-b3363023902f)
![image](https://github.com/RetroAchievements/RAWeb/assets/45672834/57c7ca08-b081-4a28-b4d9-70dd8ddf8432)
This is to make it easier to find the game since many games have localized or even censored names in other countries to make it more attractive for the regional market. Since we usually only cover the US name in the game title on our website, for Europeans like me it often happens that I need to check the internet for the US name before finding the game entry on our website.
Some devs are nice and try to cover all names in the title but this can result into a mess or sometimes is impossible because multiple titles just don't fit in the game title field.

I would like to suggest an upgrade to the game entry databases to have additional fields for alternative names, and while only one is shown as the actual game name on the game page header and game links, the others can still be found via search (and maybe can be checked on the game page itself). 

I made a small HTML example how I imagine the system, I made 10 titles in total since the example game shows there can be cases in which about these many are needed.
![image](https://github.com/RetroAchievements/RAWeb/assets/45672834/74a14926-bcf2-4069-b049-32639ccca4af)
While my example has the "Localization" as a text field, a dropdown menu might be the better idea since it's likely easier to just add a new needed entry to that menu and using an ID instead of making sure everyone follows the same writing system which would allow to improve the system further in the future like having a user setting to prioritize a specific title localizations when showing entries or having an automated official title handling like US > EU > Japanese > Others

**Comment by wescopeland** [comment]
Related https://github.com/RetroAchievements/RAWeb/discussions/741. Let's continue the conversation on this idea in this new discussion.

**Comment by Alexdatadestroyer** [comment]
Now, I imagined some things based on given idea:

If I am not mistaken, I will guess that Set Name is a single attribute on Set Table at Database. This explains, why we don't have more names as an alias for Set Name at time.

A way to allow multiple names as a Set Name would be to create another Table on Database because Set Name would have N entries. So, I wouldn't be 1:1 anymore and will go to 1:N. Based on @SporyTike example, that other Table would have ID, Name and Localization as attributes. Set ID will go as Foreign Key on that other Table in order to allow Joins. 4 attributes in total.

Regarding exhibition, since it would have multiple entries, then, an Array would be useful to show (maybe via that Help Icon) other names of the game or something that Steam does regarding username changing history (this user already played as...) when user hover on Set Name.

**Comment by Kinglink** [comment]
Going to write a dissenting opinion on this.  Personally I've been burned enough times for game name X but File needed is Japanese or European which is named Y.   Removing the expected file name from the title makes it harder to find the correct file/edition (Normally I can just look up each name). 

This sometimes is fixed by the supported Game Files page, but still can be annoying to find the correct files 

That being said, We're Back! is absolutely awful about this, and I don't have a better solution

**Comment by wescopeland** [comment]
Implemented in https://github.com/RetroAchievements/RAWeb/pull/3527.
